lib/bundler/settings.rb

```diff
@@ -214,9 +214,7 @@ def ignore_config?
 
     def app_cache_path
       @app_cache_path ||= begin
-        path = self[:cache_path] || "vendor/cache"
-        raise InvalidOption, "Cache path must be relative to the bundle path" if path.start_with?("/")
-        path
+        self[:cache_path] || "vendor/cache"
```

Comment:
I've held off extracting `"vendor/cache"` into a constant for now because it didn't appear to fit with the rest of the file. It makes sense to do but perhaps as part of a separate wider scoped refactor.